crates/core_arch/src/x86/avx512bf16.rs

```diff
@@ -0,0 +1,1122 @@
+/// corresponding mask bit is not set).
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=1769,1651,1654,1657&avx512techs=AVX512_BF16&text=_mm512_mask_cvtne2ps_pbh)
+#[inline]
+#[target_feature(enable = "avx512bf16,avx512f")]
+#[cfg_attr(test, assert_instr("vcvtne2ps2bf16"))]
+pub unsafe fn _mm512_mask_cvtne2ps_pbh (src: __m512bh, k: __mmask32, a: __m512, b: __m512) -> __m512bh {
+    let cvt = _mm512_cvtne2ps_pbh(a, b).as_u16x32();
+    transmute(simd_select_bitmask(k, cvt, src.as_u16x32()))
+}
+
+/// Convert packed single-precision (32-bit) floating-point elements in two vectors 
+/// a and b to packed BF16 (16-bit) floating-point elements, and store the results 
+/// in single vector dst using zeromask k (elements are zeroed out when the corresponding 
+/// mask bit is not set).
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=1769,1651,1654,1657&avx512techs=AVX512_BF16&text=_mm512_maskz_cvtne2ps_pbh)
+#[inline]
+#[target_feature(enable = "avx512bf16,avx512f")]
+#[cfg_attr(test, assert_instr("vcvtne2ps2bf16"))]
+pub unsafe fn _mm512_maskz_cvtne2ps_pbh (k: __mmask32, a: __m512, b: __m512) -> __m512bh {
+    let cvt = _mm512_cvtne2ps_pbh(a, b).as_u16x32();
+    let zero = _mm512_setzero_si512().as_u16x32();
+    transmute(simd_select_bitmask(k, cvt, zero))
+}
+
+/// Convert packed single-precision (32-bit) floating-point elements in a to packed BF16 (16-bit) 
+/// floating-point elements, and store the results in a 128-bit wide vector.
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=1769,1651,1654,1657,1660&avx512techs=AVX512_BF16&text=_mm_cvtneps_pbh)
+#[inline]
+#[target_feature(enable = "avx512bf16,avx512vl")]
+#[cfg_attr(test, assert_instr("vcvtneps2bf16"))]
+pub unsafe fn _mm_cvtneps_pbh (a: __m128) -> __m128bh {
+    let mut result:__m128bh;
+    asm!(
+            "vcvtneps2bf16 {1}, {0}",
+            in(xmm_reg) a,
+            lateout(xmm_reg) result,
+        );
+   result
+}
+
+/// Convert packed single-precision (32-bit) floating-point elements in a to packed BF16 (16-bit) 
+/// floating-point elements, and store the results in dst using writemask k 
+/// (elements are copied from src when the corresponding mask bit is not set). 
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=1769,1651,1654,1657,1660&avx512techs=AVX512_BF16&text=_mm_mask_cvtneps_pbh)
+#[inline]
+#[target_feature(enable = "avx512bf16,avx512vl")]
+#[cfg_attr(test, assert_instr("vcvtneps2bf16"))]
+pub unsafe fn _mm_mask_cvtneps_pbh (src: __m128bh, k:__mmask8, a: __m128) -> __m128bh {
+    let mut result:__m128bh;
+    let mask: u32 = k as u32;
+    asm!(
+            "kmovd k1, edi",
+            "vcvtneps2bf16 {1} {{k1}}, {0}",
+            in(xmm_reg) a,
+            inout(xmm_reg) src => result,
+            in("edi") mask,
```

Comment:
You need to use at&t syntax for now. Some older LLVM versions that are still supported by rustc don't correctly support intel syntax.

Reply:
Thanks @bjorn3 for your suggestion. I have updated it.